src/book/02-projects/01-nubots/03-build-system.mdx

```diff
@@ -5,3 +5,597 @@ title: Build System
+  Windows support is still in its experimental stages. Currently, Windows 10
+  Pro, Enterprise, or Education is needed to use Docker.
+</Alert>
+
+### NUClear Roles
+
+The most complete documentation for NUClear Roles can be found in its github [README](https://github.com/Fastcode/NUClearRoles/blob/master/README.md).
+
+Six main directories are used in a NUClear Roles system:
+
+| Directory        | Usage                  |
+| ---------------- | ---------------------- |
+| module           | NUClear reactors       |
+| roles            | Declaration of roles   |
+| shared/extension | NUClear DSL extensions |
+| shared/message   | Protobuf messages      |
+| shared/utility   | Utility code           |
+| tools            | Extensions for `b`     |
+
+#### Modules
+
+The modules folder contains [NUClear reactors](https://nuclear.readthedocs.io/en/latest/components.html#reactors). Reactors are grouped into meta-modules. For instance, the `vision` meta-module groups together reactors that perform vision processing tasks (e.g. ball and goal detection), while the `input` meta-module groups together reactors that gather system inputs (e.g. camera and network inputs).
+
+#### Roles
+
+Role files define which modules should be grouped together in order to form a single role (a binary that performs a specific role).
+
+Role files live in the `roles` folder and have a `.role` extension.
+
+Every role file must call the `nuclear_role()` function. Each argument to the `nuclear_role()` function is a C++ namespaced path to a module that should be included in the role. For example, if the `KinematicsConfiguration` module in `module/motion/KinematicsConfiguration` should be included in the role then `motion::KinematicsConfiguration` should appear as an argument to `nuclear_role()`.
+
+<Alert type="info">
+
+`module` is removed from the start of all module arguments as all modules must reside in the `modules` folder.
+
+</Alert>
+
+Ideally, all modules should be independent of each other and, as such, it should not matter which order modules are specified to `nuclear_role()`. In reality, there are a couple of modules which, if they are included in a role, should be passed to `nuclear_role()` first. These modules are:
+
+- `extension::FileWatcher`: Needed before all other modules so that the configuration and script systems can work properly
+- `support::SignalCatcher`: Catches exceptions to print out useful stack traces. Needs to come before all other modules to allow debugging during module installation
+- `support::logging::ConsoleLogHandler` or `support::logging::FileLogHandler`: Install before other modules so that log messages will get printed during module installation
+- `motion::KinematicsConfiguration`: Needs to come before other modules that require information about the robots kinematic configuration in order to set up properly
+
+Below is an example of a role file. Any line preceded by a `#` is a comment and will be ignored.
+
+```cmake
```

Comment:
Why is this not syntax highlighting given that you chose CMake as the language? Is there a problem with the highlighting, or is cmake not a known language. Maybe swap to `sh` if that makes the comments change colour